src/vm/virtualMachine.ts

```diff
@@ -351,4 +361,42 @@ export class VirtualMachine {
         break;
     }
   }
+
+  /** Relocate memory and trace */
+  relocate() {
+    let relocationTable = [1];
+    let memorySize = 1;
+
+    for (const segment of this.memory.values) {
+      memorySize += segment.length;
+      relocationTable.push(memorySize);
+    }
+
+    for (const [index, segment] of Object.entries(this.memory.values)) {
+      for (const [offset, value] of segment.entries()) {
+        let relocatedValue: Felt;
+        const relocatedAddr = relocationTable[Number(index)] + Number(offset);
```

Comment:
They are

Dunno why but I've used `Object.entries(this.memory.values)` (which returns a string for the key) instead of the array method `this.memory.values.entries()` to get `index` and `segment`...